sqlalchemy_sphinx/dialect.py

```diff
@@ -165,7 +159,11 @@ def check_match_clause(clause):
                 match_operators.append(clause.operator)
             elif isinstance(clause, Function):
                 if clause.name.lower() == "match":
-                    func_left, func_right = clause.clauses
+                    if len(clause.clauses) == 2:
+                        func_left, func_right = clause.clauses
+                    else:
+                        func_left = None
+                        func_right, = clause.clauses
```

Comment:
```
>>> hello = (1)
>>> left, = hello
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: cannot unpack non-iterable int object
>>> hello = (1, 2, 3)
>>> left, = hello
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: too many values to unpack (expected 1)
```

The else will always fail with a vague error not related to SQL. Throwing an exception might be the better usecase here.